packit_service/worker/handlers/testing_farm_handlers.py

```diff
@@ -102,6 +105,35 @@ def run(self) -> TaskResults:
             copr_build = CoprBuildModel.get_by_id(self.build_id)
         else:
             copr_build = testing_farm_helper.get_latest_copr_build(target=self.chroot)
+
+        # If no suitable copr build is found, run trigger copr build
+        if (
+            not copr_build
+            or copr_build.commit_sha != self.data.commit_sha
+            or copr_build.status != PG_COPR_BUILD_STATUS_SUCCESS
+        ):
+
+            logger.info("No suitable copr-build found, run copr build.")
+
+            result_details = {
+                "msg": "Build required, triggering the build",
+                "event": self.data,
+                "package_config": self.package_config,
+                "job": self.job_config.type.value if self.job_config else None,
+                "job_config": dump_job_config(self.job_config),
+            }
+
+            CoprBuildJobHelper(
+                self.service_config,
+                self.package_config,
+                self.project,
+                self.data,
+                self.db_trigger,
+                self.job_config,
+            ).run_copr_build()
```

Comment:
Just thinking out loud, what about creating a celery task? But on the other hand, this is a really simple solution,